### PR TITLE
Added a check to monitor auto scaling instance counts

### PR DESCRIPTION
**Pull request by kevinbond**
We wanted to make sure that our auto scaling instances were not out of control and wanted sensu to monitor this.

### CODE REVIEW

**Comment by sstarcher** [comment]
@kevinbond could you rebase this and refactor the aws credentials and iam roles to follow the new pattern

https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-ec2-cpu_balance.rb

File: bin/check-instances-count.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk'
+
+# Class to check the instance count
+class CheckInstanceCount < Sensu::Plugin::Check::CLI
+  option :groupname,
```
Comment (sstarcher) [inline bin/check-instances-count.rb]: Include Common here

File: bin/check-instances-count.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+         description: 'Name of the AutoScaling group',
+         short: '-g GROUP_NAME',
+         long: '--groupname GROUP_NAME',
+         required: true
+
+  option :aws_access_key,
```
Comment (sstarcher) [inline bin/check-instances-count.rb]: remove

File: bin/check-instances-count.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+         short: '-a AWS_ACCESS_KEY',
+         long: '--aws-access-key AWS_ACCESS_KEY',
+         description: "AWS Access Key. Either set ENV['AWS_ACCESS_KEY_ID'] or provide it as an option",
+         default: ENV['AWS_ACCESS_KEY_ID']
+
+  option :use_iam_role,
```
Comment (sstarcher) [inline bin/check-instances-count.rb]: remove

File: bin/check-instances-count.rb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+         short: '-u',
+         long: '--use-iam',
+         description: 'Use IAM role authenticiation. Instance must have IAM role assigned for this to work'
+
+  option :aws_secret_access_key,
```
Comment (sstarcher) [inline bin/check-instances-count.rb]: remove

File: bin/check-instances-count.rb [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+  attr_reader :aws_config, :aws_connection
+
+  def initialize
+    super
+    @aws_config = aws_login
+    @aws_connection = AWS::AutoScaling.new(@aws_config.merge!(region: config[:aws_region]))
+  end
+
+  def aws_config
```
Comment (sstarcher) [inline bin/check-instances-count.rb]: remove

File: bin/check-instances-count.rb [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    hash.update access_key_id: config[:aws_access_key], secret_access_key: config[:aws_secret_access_key]\
+      if config[:aws_access_key] && config[:aws_secret_access_key]
+    hash.update region: config[:aws_region]
+    hash
+  end
+
+  def aws_login
+    aws_config =   {}
+
+    if config[:use_iam_role].nil?
```
Comment (sstarcher) [inline bin/check-instances-count.rb]: remove

**Comment by madAndroid** [comment]
Would be awesome if this could be rebased and merged in at some point - we're looking at using this same functionality as well.

**Comment by sstarcher** [comment]
manually merged

**Pull request closed without merging**